pyphare/pyphare/core/gridlayout.py

```diff
@@ -115,26 +119,48 @@ def ghostStartIndex(self):
 
     def ghostEndIndex(self, interpOrder, centering, nbrCells):
         index = self.physicalEndIndex(interpOrder, centering, nbrCells) \
-        + self.nbrGhosts(interpOrder, centering)
+              + self.nbrGhosts(interpOrder, centering)
         return index
 
 
-    def physicalStartIndex(self,interpOrder, centering):
+    def physicalStartIndex(self, interpOrder, centering):
         index = self.ghostStartIndex() + self.nbrGhosts(interpOrder, centering)
         return index
 
 
 
     def physicalEndIndex(self, interpOrder, centering, nbrCells):
         index = self.physicalStartIndex(interpOrder, centering) \
-        + nbrCells - self.isDual(centering)
+                + nbrCells - self.isDual(centering)
         return index
 
 
 
+    def physicalStartIndices(self, qty):
+        assert qty in self.hybridQuantities
+        indices = []
+        for direction in directions[:self.box.dim()]:
+            centering = yee_centering[direction][qty]
+            indices += [self.physicalStartIndex(self.interp_order, centering)]
+        return np.asarray(indices)
+
+
+    def physicalEndIndices(self, qty):
+        assert qty in self.hybridQuantities
+        return self.physicalStartIndices(qty) + self.box.shape()
```

Comment:
-1?

Reply:
https://github.com/PHAREHUB/PHARE/blob/master/src/core/data/grid/gridlayout.h#L1331-L1337

Reply:
ok, -1 for dual